### PR TITLE
ADD: Add Lin's atmospheric science python course

### LINKED ISSUE #305: Add Johnny Lin book
Make sure the following items are specified :

- `https://www.johnny-lin.com/pycourses/`
- `An Introduction to Using Python in the Atmospheric and Oceanic Sciences`
- `This online course is a hands-on introduction to the Python programming language, tailored for atmospheric and oceanic sciences (AOS) researchers, graduate students, and advanced undergraduates who are new to Python. The course does not focus on computer science but on how Python can help AOS users do AOS work.`
- `authors` (please credit the creators of the resource in at least one method (`name`, `institution`, `institution_url`, or `email`)
- `tag` (at least one tag is required, this helps users access the link by searching via the relevant `packages`, `formats`, or `domains`)

```yaml
---
url: 'https://www.johnny-lin.com/pycourses/' # The URL link to the resource you are submitting.
thumbnail: 'https://www.johnny-lin.com/pycourses/imgs/ISS023-E-57948_lrg_200px.jpg' # The URL link to the thumbnail image associated with this resource.
title: 'An Introduction to Using Python in the Atmospheric and Oceanic Sciences' # The title of the resource.
description: 'This online course is a hands-on introduction to the Python programming language, tailored for atmospheric and oceanic sciences (AOS) researchers, graduate students, and advanced undergraduates who are new to Python. The course does not focus on computer science but on how Python can help AOS users do AOS work.' # A brief description of the resource.
authors: # Mulitiple authors are supported, add each author separately.
  - name: 'Johnny Lin' # The name of the first author.
tags: # Replace or delete the "unspecified" tag for each section with keywords associated with the resource. Multi-word tags should be entered with a space between the words (i.e. not joined with a `-`).
  formats:
    -  course # List the formats this resource takes. Currently used format tags include: book, course, documentation, gallery, tutorial, and video (but you may submit an entirely new tag if these do not best suit the resource).
---
```

### PR DESCRIPTION
**Pull request by mgrover1**
Add Johnny Lin's resource to the gallery

Closes #305

### CODE REVIEW

**Review by dcamron — DISMISSED** [review]

**Comment by dopplershift** [comment]
Are there any concerns with this being a paid resource?

**Comment by mgrover1** [comment]
There is a free version listed here
https://www.johnny-lin.com/pyintro/ed01/free_pdfs/front.pdf

File: portal/resource_gallery.yaml [hunk 1]
```diff
@@ -1491,3 +1491,17 @@
     affiliation:
     - neuromatch
     - pythia
+
+- title: An Introduction to Using Python in the Atmospheric and Oceanic Sciences
+  url: https://www.johnny-lin.com/pycourses/
```
Comment (mgrover1) [inline portal/resource_gallery.yaml:1496]:
```suggestion
  url: https://www.johnny-lin.com/pyintro/
```

**Comment by brian-rose** [comment]
Summarizing a few discussion points about this from today's EWG meeting:
- This is a historically important book for our field; many of us learned from it. However the material is very clearly out of date (e.g. predates Python 3 and key parts of the Pangeo-data stack, such as netcdf4 and xarray libraries)
- We might need to develop a policy regarding listing paid resources on our gallery, since there are *vast* numbers of paid data science learning resources out there.

**Pull request closed without merging**